Userland/Libraries/LibC/complex.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2022, Peter Elliott <pelliott@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+/* complex arithmetic
+ *
+ * https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/complex.h.html
+ */
+
+#pragma once
+
+#include <stddef.h>
+
+__BEGIN_DECLS
+
+#define complex _Complex
+
+#define _Complex_I (0.0f + 1.0fi)
+#define I _Complex_I
+
+#define CMPLX(x, y) ((double complex)__builtin_complex((double)x, (double)y))
+#define CMPLXF(x, y) ((float complex)__builtin_complex((float)x, (float)y))
+#define CMPLXL(x, y) ((long double complex)__builtin_complex((long double)x, (long double)y))
+
+float crealf(float complex z);
+double creal(double complex z);
+long double creall(long double complex z);
+
+double cimag(double complex z);
+float cimagf(float complex z);
+long double cimagl(long double complex z);
+
+#define creal(z) ((double)__real__((double complex)z))
+#define crealf(z) ((float)__real__((float complex)z))
+#define creall(z) ((long double)__real__((long double complex)z))
+
+#define cimag(z) ((double)__imag__((double complex)z))
+#define cimagf(z) ((float)__imag__((float complex)z))
+#define cimagl(z) ((long double)__imag__((long double complex)z))
```

Comment:
Instead of naming these macros the same as the functions, can you give them distinct names so `complex.cpp` doesn't require three re-reads before it's clear what it's doing? :^)

Reply:
These are supposed to be the same name. Basically we prefer that these are always inlined as macros, because there is no reason to make a function call for what is likely accessing a value in a register.

Implementing these as macros is allowed by c99:

> 7.1.4p1: Any function declared in a header may be additionally implemented as a function-like macro defined in the header.

The reason they are also declared as functions is because they need to have external linkage:

>7.1.2p6: Any declaration of a library function shall have external linkage.

Reply:
Sounds reasonable - maybe add some of this reasoning to the commit message or as a comment?

Reply:
comments added